app/controllers/application_controller.rb

```diff
@@ -165,6 +165,13 @@ def generate_short(msg)
   end
   helper_method :generate_short
 
+  def readonly_forbidden
+    return unless logged_in?
+    return unless current_user.readonly?
+    flash[:error] = "This feature is not available to read-only accounts."
+    redirect_to continuities_path
```

Comment:
Hmm on one hand a better redirect might be good, on the other that sounds potentially tricky